### PR TITLE
8263461: jdk/jfr/event/gc/detailed/TestEvacuationFailedEvent.java uses wrong mechanism to cause evacuation failure

### PR DESCRIPTION
**Pull request by tschatzl**
Hi all,

  can I have reviews for this change to the jdk/jfr/event/gc/detailed/TestEvacuationFailedEvent.java test to use the only "officially sanctioned" way to cause evacuation failures via the respective debug-level options.

Testing: checking failing test case to cause evacuation failures

Thanks,
  Thomas

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8263461](https://bugs.openjdk.java.net/browse/JDK-8263461): jdk/jfr/event/gc/detailed/TestEvacuationFailedEvent.java uses wrong mechanism to cause evacuation failure

### Reviewers
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**)
 * [Ivan Walulya](https://openjdk.java.net/census#iwalulya) (@walulyai - Committer)
 * [Albert Mingkun Yang](https://openjdk.java.net/census#ayang) (@albertnetymk - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4541/head:pull/4541` \
`$ git checkout pull/4541`

Update a local copy of the PR: \
`$ git checkout pull/4541` \
`$ git pull https://git.openjdk.java.net/jdk pull/4541/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4541`

View PR using the GUI difftool: \
`$ git pr show -t 4541`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4541.diff">https://git.openjdk.java.net/jdk/pull/4541.diff</a>

</details>

### CODE REVIEW

File: test/jdk/jdk/jfr/event/gc/detailed/TestEvacuationFailedEvent.java [hunk 1]
```diff
@@ -23,43 +23,58 @@
...
-            return;
+        Object[] data = new Object[1024];
+
+        for (int i = 0; i < data.length; i++) {
+            data[i] = new byte[5 * 1024];
         }
+        // Guarantee one young gc.
+        WhiteBox.getWhiteBox().youngGC();
+        // Keep alive data.
+        System.out.println(data);
```
Comment (kimbarrett) [inline test/jdk/jdk/jfr/event/gc/detailed/TestEvacuationFailedEvent.java:72]: `Reference.reachabilityFence` seems like a better choice for keep-alive.  Other than that, looks good.

**Review by kimbarrett — CHANGES_REQUESTED** [review]

**Review by kimbarrett — APPROVED** [review]
Looks good.

**Review by walulyai — APPROVED** [review]
lgtm!

**Review by albertnetymk — APPROVED** [review]

**Comment by tschatzl** [comment]
Thanks @kimbarrett @walulyai @albertnetymk for your reviews.

/integrate

**Pull request closed without merging**